### PR TITLE
Vulkan Guide draft entry for VK_KHR_shader_subgroup_uniform_control_flow

### PR DESCRIPTION
**Pull request by alan-baker**

### CODE REVIEW

File: chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# VK_KHR_shader_subgroup_uniform_control_flow
+
+## Overview
+
+VK_KHR_shader_subgroup_uniform_control provides stronger guarantees for
+reconvergence of invocations in a shader. If the extension is supported,
+shaders can be modified to include a new attribute that provides the stronger
+guarantees (see GL_EXT_subgroup_uniform_control_flow). This attribute can only
```
Comment (sfricke-samsung) [inline chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md:8]:
```suggestion
guarantees (see [GL_EXT_subgroup_uniform_control_flow](https://github.com/KhronosGroup/GLSL/blob/master/extensions/ext/GL_EXT_subgroup_uniform_control_flow.txt)). This attribute can only
```

File: chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+```
+
+There is a problem with the code that might lead to unexpected results. Vulkan
+only requires invocations to reconverge after the if statement that performs
+the subgroup election if all the invocations in the __workgroup__ are converged at
+that if statement. If the invocations don’t reconverge then the broadcast and
+offset calculations will be incorrect. Not all invocations would write their
+results to the correct index.
+
+VK_KHR_shader_subgroup_uniform_control flow can be utilized to make the shader
```
Comment (sfricke-samsung) [inline chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md:64]:
```suggestion
`VK_KHR_shader_subgroup_uniform_control_flow` can be utilized to make the shader
```

File: chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
+# VK_KHR_shader_subgroup_uniform_control_flow
+
+## Overview
+
+VK_KHR_shader_subgroup_uniform_control provides stronger guarantees for
```
Comment (sfricke-samsung) [inline chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md:5]:
```suggestion
[VK_KHR_shader_subgroup_uniform_control_flow](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_KHR_shader_subgroup_uniform_control_flow.html) provides stronger guarantees for
```

File: chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+reconvergence of invocations in a shader. If the extension is supported,
+shaders can be modified to include a new attribute that provides the stronger
+guarantees (see GL_EXT_subgroup_uniform_control_flow). This attribute can only
+be applied to shader stages that support subgroup operations (check the
+subgroupSupportedStages feature in the VkPhysicalDeviceVulkan11Properties
```
Comment (sfricke-samsung) [inline chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md:10]:
```suggestion
`subgroupSupportedStages` feature in the `VkPhysicalDeviceVulkan11Properties`
```

File: chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      // requested space.
+      b.data[base + offset] = ...;
+    }
+  }
+  ...
+}
+```
+
+The differences from the original shader are relatively minor. First, the
+addition of the subgroup_uniform_control_flow attribute informs the
```
Comment (sfricke-samsung) [inline chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md:105]:
```suggestion
addition of the `subgroup_uniform_control_flow` attribute informs the
```

**Review by sfricke-samsung — CHANGES_REQUESTED** [review]
Just some markdown nits and should be ready to go

File: chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md (hunk 4 above)
Comment (SaschaWillems) [inline chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md:10]: Afaik `VkPhysicalDeviceVulkan11Properties` requires Vulkan 1.2, but the control flow extension is written against Vulkan 1.1. for which `VkPhysicalDeviceSubgroupProperties` can be used to query the supported stages. Not sure if that's too much detail though.

Comment (sfricke-samsung) [reply chapters/extensions/VK_KHR_shader_subgroup_uniform_control_flow.md:10]: Could just say:

Check `VkPhysicalDeviceSubgroupProperties::supportedStages` or `VkPhysicalDeviceVulkan11Properties::subgroupSupportedStages`

**Comment by sfricke-samsung** [comment]
Once https://github.com/KhronosGroup/GLSL/pull/160 is merged will merge this as URL is broken otherwise (what is failing CI)

**Comment by alan-baker** [comment]
@sfricke-samsung the GLSL extension was merged.

**Pull request merged**